packages/protocol/contracts/common/GasPriceMinimum.sol

```diff
@@ -108,25 +115,45 @@ contract GasPriceMinimum is
     emit GasPriceMinimumFloorSet(_gasPriceMinimumFloor);
   }
 
+  /**
+   * @notice Set the activation block of the baseFee opCode.
+   * @param _baseFeeOpCodeActivationBlock Block number where the baseFee opCode is activated
+   * @dev Value is expected to be > 0.
+   */
+  function setBaseFeeOpCodeActivationBlock(uint256 _baseFeeOpCodeActivationBlock) public onlyOwner {
+    require(
+      _baseFeeOpCodeActivationBlock > 0,
+      "baseFee opCode activation block must be greater than zero"
+    );
+    baseFeeOpCodeActivationBlock = _baseFeeOpCodeActivationBlock;
+    emit BaseFeeOpCodeActivationBlockSet(_baseFeeOpCodeActivationBlock);
+  }
+
   /**
    * @notice Retrieve the current gas price minimum for a currency.
    * @param tokenAddress The currency the gas price should be in (defaults to gold).
    * @return current gas price minimum in the requested currency
    */
   function getGasPriceMinimum(address tokenAddress) external view returns (uint256) {
+    uint256 _gasPriceMinimum;
+    if (baseFeeOpCodeActivationBlock > 0 && block.number >= baseFeeOpCodeActivationBlock) {
+      _gasPriceMinimum = block.basefee;
+    } else {
+      _gasPriceMinimum = gasPriceMinimum;
+    }
```

Comment:
are we planning to add a test case that covers this?

Reply:
Yes, I'm waiting for the 0.8.7 version resolution, of if we switch strategies